### ISSUE TITLE
Issue using TS with 36.5.0

### ISSUE THREAD

**Issue by ssandoy** [FormidableLabs/victory#2311]
# Bug Reports

### Checklist

- [ ] I have read through the [FAQ](https://formidable.com/open-source/victory/docs/faq) and [Guides](https://formidable.com/open-source/victory/guides/)

- [x] I am using the latest version of Victory

- [ ] I've searched open issues to make sure I'm not opening a duplicate issue

### The Problem

My TS-based project is unable to locate the types defined in victory-core when running `tsc` in my project. 
I get the `error TS2305: Module '"victory-core"' has no exported member 'StringOrNumberOrCallback'.` 
By inspecting the code I suspect that the lack of explicit export of the [types.ts](https://github.com/FormidableLabs/victory/blob/v36.5.0/packages/victory-core/src/types.ts) in [victory-core/index.ts](https://github.com/FormidableLabs/victory/blob/v36.5.0/packages/victory-core/src/index.ts) might be the issue?

### Reproduction
Running `tsc` in a project using TypeScript when importing victory-packages.

# Feature Requests

### Checklist

- [x] I've read through the [Docs](https://formidable.com/open-source/victory/docs) and [Guides](https://formidable.com/open-source/victory/guides) to make sure this functionality doesn't already exist

- [x] I've searched open issues to make sure I'm not opening a duplicate issue

### Description

Make the TypeScript happy again by adding the export of the types.ts-file inside victory-core.

**Comment by kubik369** [comment]
I have encountered the same problem. I have been able to make my `tsc` happy by doing:
```typescript
import {VictoryStringOrNumberCallback} from 'victory-core/lib/types'
```
instead of
```typescript
import {VictoryStringOrNumberCallback} from 'victory-core'
```
Not perfect, but from looking at PRs, authors are already working on making this better, so it appears to be a good workaround for the time being. If it is not ok, please let me know, I don't have that much experience regarding this :/

**Comment by becca-bailey** [comment]
We will check on this and make sure the correct type is exported from victory-core.

**Comment by becca-bailey** [comment]
By the way, your workaround should work for now! Apologies for the confusion, we're in the middle of a typescript conversion so things might be getting moved around unintentionally. Everything looks right on my end, but I will investigate a little more.

**Comment by becca-bailey** [comment]
This should be resolved if you upgrade to 36.5.1.

**Comment by ssandoy** [comment]
> We currently have this fixed in `main`, but haven't published this version yet. Once we publish we'll update this ticket. Thanks for reporting it!

I tried tou pdate to 36.5.2, and that seemed to remove the compiler-errors related to the types in victory-core. 
I get a number of other errors now, however:

```
error TS2430: Interface 'VictoryVoronoiProps' incorrectly extends interface 'VictoryDatableProps'.
  Types of property 'sortKey' are incompatible.
    Type 'string[] | StringOrNumberOrCallback | undefined' is not assignable to type 'DataGetterPropType | undefined'.
      Type 'VictoryStringOrNumberCallback' is not assignable to type 'DataGetterPropType | undefined'.
        Type 'VictoryStringOrNumberCallback' is not assignable to type '(props: object) => string | number | string[] | number[]'.
          Types of parameters 'args' and 'props' are incompatible.
            Property 'index' is missing in type '{}' but required in type 'CallbackArgs'.

error TS2305: Module '"victory-brush-container"' has no exported member 'BrushHelpers'.
error TS2305: Module '"victory-brush-container"' has no exported member 'brushContainerMixin'.
error TS2305: Module '"victory-cursor-container"' has no exported member 'CursorHelpers'.
error TS2305: Module '"victory-cursor-container"' has no exported member 'cursorContainerMixin'.
error TS2305: Module '"victory-selection-container"' has no exported member 'SelectionHelpers'.
error TS2305: Module '"victory-selection-container"' has no exported member 'selectionContainerMixin'.
error TS2305: Module '"victory-voronoi-container"' has no exported member 'VoronoiHelpers'.
error TS2305: Module '"victory-voronoi-container"' has no exported member 'voronoiContainerMixin'.
error TS2305: Module '"victory-zoom-container"' has no exported member 'ZoomHelpers'.
error TS2305: Module '"victory-zoom-container"' has no exported member 'zoomContainerMixin'.
error TS2305: Module '"victory-zoom-container"' has no exported member 'RawZoomHelpers'.
error TS2305: Module '"victory-create-container"' has no exported member 'combineContainerMixins'.
error TS2305: Module '"victory-create-container"' has no exported member 'makeCreateContainerFunction'.
error TS2305: Module '"victory-canvas"' has no exported member 'useCanvasContext'.
```
